[all hosts] (Publishing) Guidance around add-ins in government cloud environments (#3699)
OfficeDev/office-js-docs-pr: docs/publish/government-cloud-guidance.md

For partners providing their Office Add-ins to customers in these restricted cloud environments, there are important differences from the standard public cloud environment that must be considered. The following information gives the details that require special handling by developers writing Office Add-ins that target customers in these environments.

## All sovereign environments

For all government cloud (i.e. Sovereign Cloud) environments, the public Office Store is not available. This means that end-users are not able to acquire Office Add-ins directly from the public store. Administrators are also unable to deploy Office Add-ins directly from the public store into their Admin Portal. Instead, you must work with administrators to ensure the following:

- The required resources and services for your solution are available inside the cloud boundary. This means that you have either worked with the tenant administrators to provision your service and resources inside of the cloud boundary, or you have worked with the network administrator to enable access to the your resources that reside outside of the cloud boundary.

> - The resources being accessed through the Office Add-in conform to the requirements of the government cloud from which they are being accessed. They also must conform to any additional requirements from the customer tenant for which the solution is being provisioned with regards to the transfer, management, and storage of potentially sensitive data, as well as code and resource security and access vetting procedures.

Reviewer: ```suggestion
- The resources the Office Add-in accesses conform to the requirements of the government cloud from which they are being accessed. They also must conform to any additional requirements from the customer tenant for which the solution is being provisioned. These requirements include the transfer, management, and storage of potentially sensitive data, as well as code and resource security and access vetting procedures.

- The Office Add-in manifest that describes the solution and its source location as applicable to the particular government cloud deployment is obtained from the partner and ingested for deployment to the appropriate users via the Admin Portal.

## US Government Community Cloud (GCC)

> In addition to the above requirements applicable to all Sovereign Clouds, each Sovereign Cloud environment has its own considerations that may affect Office Add-ins targeting the environment. GCC is the least restrictive of the government cloud environments and some of the resources required by solutions built on top of Office are available from their usual production endpoints in this environment. One such resource is the office.js library. Solution partners should be able to continue to reference the public office.js resource as they do with their public production solution.

Reviewer: ```suggestion
In addition to requirements applicable to all Sovereign Clouds, each Sovereign Cloud environment has its own considerations that may affect Office Add-ins targeting the environment. GCC is the least restrictive of the government cloud environments and some of the resources required by the add-in are available from their usual production endpoints in this environment. One such resource is the Office JavaScript API library. Solution partners can continue to reference the public Office.js resource as they do with their public production solution.
```

Author: I want to be careful about how we refer to office.js here, since it's a specific resource. @jeremy-msft, what's the right phrasing here?

Reviewer: I verified in the docs that we're using Office.js capitalized (except in code) and so I went with that. Here are two examples

![image](https://user-images.githubusercontent.com/30682803/196298102-50fd4b11-8a74-4e9c-9c1d-c084caac1308.png)

Author: Cool, thanks @lindalu-MSFT. I'll make those changes.

## GCC High (GCCH), US Department of Defense (DOD), or other sovereign managed clouds

> These government clouds remain internet-connected, but the set of public endpoints made available from these clouds is severely restricted. One such restricted endpoint is the public endpoint for loading the office.js library. The public CDN location for this resource will not be accessible from within these environments. However, there will be an internal per cloud Microsoft Office CDN provisioned with the same resource. This means the endpoint URL to access this resource will be different within these environments. Your Office Add-in may need some level of customization to run within these environments. Given the additional restrictions in these environments, it's likely that any solution provided to customers in these clouds will require hosting provider services within the environment as well, necessitating additional customizations beyond the access of the office.js library. You'll need to determine the best way to provide your solution to customers in these environments, such that it conforms to the additional restrictions these environments impose on services running within their boundaries.

Reviewer: ```suggestion
These government clouds remain internet-connected, but the set of public endpoints made available is severely restricted. One such restriction is the public endpoint for loading the Office.js library. The public CDN location for Office.js will not be accessible from within these environments. However, there will be an internal per cloud Microsoft Office CDN provisioned with the same resource. This means the endpoint URL to access Office.js will be different and your Office Add-in may need some level of customization to run. Given the additional restrictions, it's likely that any solution provided to customers will require hosting provider services within the environment as well, necessitating additional customizations. You'll need to determine the best way to provide your solution to customers, such that it conforms to the additional restrictions imposed on services running within the boundaries of these environments.
```

Reviewer: Not sure why edits are not highlighted in this suggestion.

> ## Airgapped sovereign clouds

Reviewer: ```suggestion
## Airgapped Sovereign Clouds
```

> These government clouds are essentially disconnected from the public internet entirely. Any resource that would normally be accessed from public resources must instead be custom-provisioned inside these cloud environments. In the GCCH and DOD clouds mentioned earlier, while most (if not all) solution providers will need to provision their services and resources inside the cloud, there is an option to make firewall exceptions in these environments that allows access to public services and resources. This bypass is not possible in airgapped clouds. As with the GCCH and DOD clouds, there will be a Microsoft Office CDN provisioned inside each cloud environment that hosts required resources such as the office.js library. However, you'll need to work closely with customer tenant administrators to determine the best way to provide your services and resources in a way that conforms to the strict access requirements for these clouds.

Reviewer: ```suggestion
These government clouds are essentially disconnected from the public internet entirely. Any resource that would normally be accessed from public resources must instead be custom-provisioned inside these cloud environments. In the GCCH and DOD clouds mentioned previously, most (if not all) solution providers will need to provision their services and resources inside the cloud. There is an option to make firewall exceptions that allows access to public services and resources. However, this bypass is not possible in airgapped clouds. As with the GCCH and DOD clouds, there will be a Microsoft Office CDN provisioned inside each cloud environment that hosts required resources such as the Office.js library. You'll need to work closely with customer tenant administrators to determine the best way to provide your services and resources in a way that conforms to the strict access requirements for Airgapped Sovereign Clouds.
```

## See also

> - [Onboarding to the Office 365 U.S. Government Clouds](https://govprograms.office.net/content/onboarding/overview.html)

Reviewer: ```suggestion
- [Onboard to the Office 365 U.S. Government clouds](https://govprograms.office.net/content/onboarding/overview.html)
```

Author: This is the title from the linked page, so I'll leave it as is to match.

Reviewer: True, but the TOC reference is Onboard. Someone on their end can fix the inconsistency.

![image](https://user-images.githubusercontent.com/30682803/196298400-c2af0b88-6e69-482e-a416-53013ab46886.png)
